// @theme-ui/preset-future/dist/index.umd.js
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@theme-ui/preset-base')) :
  typeof define === 'function' && define.amd ? define(['exports', '@theme-ui/preset-base'], factory) :
  (global = global || self, factory(global.presetFuture = {}, global.base));
}(this, (function (exports, base) {
  function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

  var base__default = /*#__PURE__*/_interopDefaultLegacy(base);

  function _extends() {
    _extends = Object.assign || function (target) {
      for (var i = 1; i < arguments.length; i++) {
        var source = arguments[i];

        for (var key in source) {
          if (Object.prototype.hasOwnProperty.call(source, key)) {
            target[key] = source[key];
          }
        }
      }

      return target;
    };

    return _extends.apply(this, arguments);
  }

  var future = _extends({}, base__default['default'], {
    colors: {
      text: '#000',
      background: '#fff',
      primary: '#11e',
      secondary: '#c0c',
      highlight: '#e0e',
      muted: '#f6f6ff',
      modes: {
        dark: {
          text: '#fff',
          background: '#000',
          primary: '#0fc',
          secondary: '#0cf',
          highlight: '#f0c',
          muted: '#011'
        }
      }
    },
    fonts: {
      body: '"Avenir Next", system-ui, sans-serif',
      heading: 'inherit',
      monospace: 'Menlo, monospace'
    },
    fontWeights: {
      body: 400,
      heading: 600,
      bold: 700
    },
    lineHeights: {
      body: 1.75,
      heading: 1.25
    }
  });

  exports.default = future;
  exports.future = future;

})));
