// @theme-ui/preset-polaris/dist/index.umd.js
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@theme-ui/preset-base')) :
  typeof define === 'function' && define.amd ? define(['exports', '@theme-ui/preset-base'], factory) :
  (global = global || self, factory(global.presetPolaris = {}, global.base));
}(this, (function (exports, base) {
  function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

  var base__default = /*#__PURE__*/_interopDefaultLegacy(base);

  function _extends() {
    _extends = Object.assign || function (target) {
      for (var i = 1; i < arguments.length; i++) {
        var source = arguments[i];

        for (var key in source) {
          if (Object.prototype.hasOwnProperty.call(source, key)) {
            target[key] = source[key];
          }
        }
      }

      return target;
    };

    return _extends.apply(this, arguments);
  }

  var text = 'system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", sans-serif';
  var polaris = _extends({}, base__default['default'], {
    colors: {
      text: '#454f5b',
      background: '#fff',
      primary: '#5c6ac4',
      secondary: '#006fbb',
      highlight: '#47c1bf',
      muted: '#e6e6e6',
      gray: '#dfe3e8',
      accent: '#f49342',
      darken: '#00044c',
      modes: {
        dark: {
          text: '#3e4155',
          background: '#000639',
          primary: '#9c6ade',
          secondary: '#b4e1fa',
          highlight: '#b7ecec',
          muted: '#e6e6e6'
        }
      }
    },
    fonts: {
      body: text,
      heading: text,
      monospace: 'Menlo, monospace'
    },
    fontWeights: {
      body: 400,
      heading: 600,
      bold: 700
    },
    lineHeights: {
      body: 1.75,
      heading: 1.25
    }
  });

  exports.default = polaris;
  exports.polaris = polaris;

})));
