// @theme-ui/preset-funk/dist/index.umd.js
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@theme-ui/preset-base')) :
  typeof define === 'function' && define.amd ? define(['exports', '@theme-ui/preset-base'], factory) :
  (global = global || self, factory(global.presetFunk = {}, global.base));
}(this, (function (exports, base) {
  function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

  var base__default = /*#__PURE__*/_interopDefaultLegacy(base);

  function _extends() {
    _extends = Object.assign || function (target) {
      for (var i = 1; i < arguments.length; i++) {
        var source = arguments[i];

        for (var key in source) {
          if (Object.prototype.hasOwnProperty.call(source, key)) {
            target[key] = source[key];
          }
        }
      }

      return target;
    };

    return _extends.apply(this, arguments);
  }

  var funk = _extends({}, base__default['default'], {
    fonts: {
      body: 'Poppins, sans-serif',
      heading: 'Poppins, sans-serif',
      monospace: 'Menlo, monospace'
    },
    lineHeights: {
      body: 1.625,
      heading: 1.25
    },
    fontWeights: {
      body: 400,
      heading: 900,
      bold: 700
    },
    colors: _extends({}, base__default['default'].colors, {
      primary: '#609',
      secondary: '#306'
    }),
    styles: _extends({}, base__default['default'].styles)
  });

  exports.default = funk;
  exports.funk = funk;

})));
