// @theme-ui/presets/dist/index.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@theme-ui/preset-base'), require('@theme-ui/preset-dark'), require('@theme-ui/preset-deep'), require('@theme-ui/preset-funk'), require('@theme-ui/preset-future'), require('@theme-ui/preset-roboto'), require('@theme-ui/preset-swiss'), require('@theme-ui/preset-system'), require('@theme-ui/preset-tosh'), require('@theme-ui/preset-bootstrap'), require('@theme-ui/preset-bulma'), require('@theme-ui/preset-polaris'), require('@theme-ui/preset-tailwind'), require('@theme-ui/preset-sketchy')) :
	typeof define === 'function' && define.amd ? define(['exports', '@theme-ui/preset-base', '@theme-ui/preset-dark', '@theme-ui/preset-deep', '@theme-ui/preset-funk', '@theme-ui/preset-future', '@theme-ui/preset-roboto', '@theme-ui/preset-swiss', '@theme-ui/preset-system', '@theme-ui/preset-tosh', '@theme-ui/preset-bootstrap', '@theme-ui/preset-bulma', '@theme-ui/preset-polaris', '@theme-ui/preset-tailwind', '@theme-ui/preset-sketchy'], factory) :
	(global = global || self, factory(global.presets = {}, global.presetBase, global.presetDark, global.presetDeep, global.presetFunk, global.presetFuture, global.presetRoboto, global.presetSwiss, global.presetSystem, global.presetTosh, global.presetBootstrap, global.presetBulma, global.presetPolaris, global.presetTailwind, global.presetSketchy));
}(this, (function (exports, presetBase, presetDark, presetDeep, presetFunk, presetFuture, presetRoboto, presetSwiss, presetSystem, presetTosh, presetBootstrap, presetBulma, presetPolaris, presetTailwind, presetSketchy) {
	function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

	var presetBase__default = /*#__PURE__*/_interopDefaultLegacy(presetBase);
	var presetDark__default = /*#__PURE__*/_interopDefaultLegacy(presetDark);
	var presetDeep__default = /*#__PURE__*/_interopDefaultLegacy(presetDeep);
	var presetFunk__default = /*#__PURE__*/_interopDefaultLegacy(presetFunk);
	var presetFuture__default = /*#__PURE__*/_interopDefaultLegacy(presetFuture);
	var presetRoboto__default = /*#__PURE__*/_interopDefaultLegacy(presetRoboto);
	var presetSwiss__default = /*#__PURE__*/_interopDefaultLegacy(presetSwiss);
	var presetSystem__default = /*#__PURE__*/_interopDefaultLegacy(presetSystem);
	var presetTosh__default = /*#__PURE__*/_interopDefaultLegacy(presetTosh);
	var presetBootstrap__default = /*#__PURE__*/_interopDefaultLegacy(presetBootstrap);
	var presetBulma__default = /*#__PURE__*/_interopDefaultLegacy(presetBulma);
	var presetPolaris__default = /*#__PURE__*/_interopDefaultLegacy(presetPolaris);
	var presetTailwind__default = /*#__PURE__*/_interopDefaultLegacy(presetTailwind);
	var presetSketchy__default = /*#__PURE__*/_interopDefaultLegacy(presetSketchy);



	Object.defineProperty(exports, 'base', {
		enumerable: true,
		get: function () {
			return presetBase__default['default'];
		}
	});
	Object.defineProperty(exports, 'dark', {
		enumerable: true,
		get: function () {
			return presetDark__default['default'];
		}
	});
	Object.defineProperty(exports, 'deep', {
		enumerable: true,
		get: function () {
			return presetDeep__default['default'];
		}
	});
	Object.defineProperty(exports, 'funk', {
		enumerable: true,
		get: function () {
			return presetFunk__default['default'];
		}
	});
	Object.defineProperty(exports, 'future', {
		enumerable: true,
		get: function () {
			return presetFuture__default['default'];
		}
	});
	Object.defineProperty(exports, 'roboto', {
		enumerable: true,
		get: function () {
			return presetRoboto__default['default'];
		}
	});
	Object.defineProperty(exports, 'swiss', {
		enumerable: true,
		get: function () {
			return presetSwiss__default['default'];
		}
	});
	Object.defineProperty(exports, 'system', {
		enumerable: true,
		get: function () {
			return presetSystem__default['default'];
		}
	});
	Object.defineProperty(exports, 'tosh', {
		enumerable: true,
		get: function () {
			return presetTosh__default['default'];
		}
	});
	Object.defineProperty(exports, 'bootstrap', {
		enumerable: true,
		get: function () {
			return presetBootstrap__default['default'];
		}
	});
	Object.defineProperty(exports, 'bulma', {
		enumerable: true,
		get: function () {
			return presetBulma__default['default'];
		}
	});
	Object.defineProperty(exports, 'polaris', {
		enumerable: true,
		get: function () {
			return presetPolaris__default['default'];
		}
	});
	Object.defineProperty(exports, 'tailwind', {
		enumerable: true,
		get: function () {
			return presetTailwind__default['default'];
		}
	});
	Object.defineProperty(exports, 'sketchy', {
		enumerable: true,
		get: function () {
			return presetSketchy__default['default'];
		}
	});

})));
